AWSAppSyncClient/Internal/AWSPerformMutationQueue.swift

```diff
@@ -162,9 +162,9 @@ final class AWSPerformMutationQueue {
             .catch { error in AppSyncLog.error("\(#function) failure: \(error)") }
     }
 
-    func clearQueue() {
+    func clearQueue() throws {
         operationQueue.cancelAllOperations()
-        _ = persistentCache?.clear()
+        try persistentCache?.clear().await()
```

Comment:
same as above. Need to ensure promise is not resolved to error case.

Reply:
await should throw in failure case